docs/source/operators/security.rst

```diff
@@ -77,17 +77,130 @@ but this is **NOT RECOMMENDED**, unless authentication or access restrictions ar
     c.ServerApp.token = ''
     c.ServerApp.password = ''
 
-Authorization
--------------
+
+Authentication and Authorization
+--------------------------------
 
 .. versionadded:: 2.0
 
+There are two steps to deciding whether to allow a given request to be happen.
+
+The first step is "Authentication" (identifying who is making the request).
+This is handled by the :class:`.IdentityProvider`.
+
+Whether a given user is allowed to take a specific action is called "Authorization",
+and is handled separately, by an :class:`.Authorizer`.
+
+These two classes may work together,
+as the information returned by the IdentityProvider is given to the Authorizer when it makes its decisions.
+
+Authentication always takes precedence because if no user is authenticated,
+no authorization checks need to be made,
+as all requests requiring _authorization_ must first complete _authentication_.
+
+Identity Providers
+******************
+
+The :class:`.IdentityProvider` class is responsible for the "authorization" step,
+identifying the user making the request,
+and constructing information about them.
+
+It principally implements two methods.
+
+.. autoclass:: jupyter_server.auth.IdentityProvider
+
+  .. automethod:: get_user
+  .. automethod:: identity_model
+
+The first is :meth:`.IdentityProvider.get_user(Handler)`.
+This method is given a RequestHandler, and is responsible for deciding whether there is an authenticated user making the request.
+If the request is authenticated, it should return a truthy object representing the authenticated user.
+It should return None if the request is not authenticated.
+
+The default implementation accepts token or password authentication
+
+This object will be available as `self.current_user` in any request handler.
+Request methods decorated with tornado's `@web.authenticated` decorator.
+will only be allowed if this method returns something.
```

Comment:
Trailing period in above line. Also: Maybe we should link to the tornado docs here (https://www.tornadoweb.org/en/stable/web.html#tornado.web.authenticated) and/or explicitly state what it will do ("If the user is not logged in, they will be redirected to the configured [login url](https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler.get_login_url).")